app/services/sync_provider_from_find.rb

```diff
@@ -0,0 +1,38 @@
+class SyncProviderFromFind
+  def self.call(provider_code:)
+    find_provider = FindAPI::Provider
+      .current_cycle
+      .includes(:sites, courses: [:sites])
+      .find(provider_code)
+      .first
+    find_courses = find_provider.courses
+    find_sites = find_provider.sites
+
+    provider = Provider.find_or_create_by(code: provider_code)
+    provider.name = find_provider.provider_name
+    provider.save
+
+    sites = find_sites.map do |find_site|
+      site = provider.sites.find_or_create_by(code: find_site.location_name)
+      site.name = find_site.location_name
+      site.save
+      site
+    end
```

Comment:
This is complex to follow (for me at least :P), the map should just return an array, it'd be better to split this into a `.each` and then a simple `map`, if you could use private method for this would be great.
(I know it is a service, we could use `class << self` if we need more methods)

Reply:
:+1: I'll see if I can clean this up a bit

Reply:
Much better now! thanks
as I said, if you want you can group the `self.____` methods using the pattern `class << self`, and you can make `private` on those methods that you do not need to expose.